p2p/libp2p/netMessenger.go

```diff
@@ -1100,7 +1103,11 @@ func (netMes *networkMessenger) GetConnectedPeersInfo() *p2p.ConnectedPeersInfo
 			if selfPeerInfo.ShardID != peerInfo.ShardID {
 				connPeerInfo.CrossShardObservers[peerInfo.ShardID] = append(connPeerInfo.CrossShardObservers[peerInfo.ShardID], connString)
 			} else {
-				connPeerInfo.IntraShardObservers[peerInfo.ShardID] = append(connPeerInfo.IntraShardObservers[peerInfo.ShardID], connString)
+				if peerInfo.PeerSubType == core.FullHistoryObserver {
```

Comment:
too many nested conditions, can we change it to something like:
```go
case core.ObserverPeer:
    connPeerInfo.NumObservers[peerInfo.ShardID]++
    if selfPeerInfo.ShardID != peerInfo.ShardID {
        connPeerInfo.CrossShardObservers[peerInfo.ShardID] = append(connPeerInfo.CrossShardObservers[peerInfo.ShardID], connString)
        break
    }
    if peerInfo.PeerSubType == core.FullHistoryObserver {
        connPeerInfo.FullHistoryObservers[peerInfo.ShardID] = append(connPeerInfo.FullHistoryObservers[peerInfo.ShardID], connString)
        break
    }

    connPeerInfo.IntraShardObservers[peerInfo.ShardID] = append(connPeerInfo.IntraShardObservers[peerInfo.ShardID], connString)
```
then we do not need any else clauses.

Reply:
Done